src/Utilities/OperatingSystemFilterHelpers.js

```diff
@@ -89,3 +87,18 @@ export const buildOSChip = (operatingSystemValue = {}, operatingSystems) => {
         ]
         : [];
 };
+
+export const onOSFilterChange = (event, selection, clickedGroup, clickedItem) => {
+    const newSelection = Object.assign({}, selection);
+    const value = newSelection[clickedGroup.value][clickedItem.value];
+    const major = clickedGroup.value;
+
+    if (clickedItem.value === major) {
+        // group checkbox clicked => update all minor version selections
+        newSelection[major] = mapValues(newSelection[major], () => value);
+    } else {
+        newSelection[major][major] = Object.values(omit(newSelection[major], major)).every(Boolean);
```

Comment:
Can we avoid having an object property with the an inner property with the same name?

Reply:
@mkholjuraev , why?

Reply:
There might be some benefits of a current object that you are aware of, but I am not. However, for example, this seems to be not a very clean object. 

`const object = {  objectKey: { objectKey: anotherValue } }`
 Ignore if this does not make sense or if changing this takes a lot of effort

Reply:
The thing is, I needed to come up with some kind of data structure that could reflect all the selection cases also given the property format of PrimaryToolbar... I mean, yeah, this is complex and without the context this looks totally weird, but I tried my best to write this business logic explicitly and some comments should also be suggesting why this works like that...